src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -289,35 +303,19 @@ private boolean runRepair() {
       }
 
       try (Timer.Context cxt1 = context.metricRegistry.timer(metricNameForRepairing(segment)).time()) {
-        boolean segmentsLocked = false;
+        //boolean segmentsLocked = false;
         try {
           LOG.debug("Enter synchronized section with segment ID {}", segmentId);
           synchronized (condition) {
```

Comment:
is the `synchronized` block needed anymore? is it only relevant to non-distributed mode now?

Reply:
Reaper can have multiple threads accessing the same segments (such as the one looking for segments with no leader, etc...). The JMX notifications can also lead to out of order updates without the synchronized blocks, so we'd better keep them as is.